src/Model/BlogObject.php

```diff
@@ -72,6 +72,10 @@ public function validate()
             $validation->addError($this->getDuplicateError(), self::DUPLICATE_EXCEPTION);
         }
 
+        if (empty($this->Title)) {
+            $validation->addError($this->getEmptyTitleError(), self::EMPTY_TITLE_EXCEPTION);
```

Comment:
```suggestion
            $validation->addError($this->getEmptyTitleError(), self::EMPTY_TITLE_EXCEPTION ?? 'EMPTY_TITLE');
```
The constant won't be declared on any custom classes that use this trait, and requiring it to be added is technically a breaking change.

Note: Please test that this actually works - I've not tried the null coalescing operator in this context before. I think it'll work but it'll pay to check.

Reply:
Yikes, shouldn't be calling self::<const> on a trait.  Not a great choice when this was made back in the day

I don't think null coalescing works here

```php
class C { function F(){ var_dump(self::Z ?? 'default'); } }; (new C)->F();
// PHP Warning:  Uncaught Error: Undefined constant C::Z in php shell cod
```